firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -180,44 +177,51 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
   @NonNull
   public UpdateTask updateApp() throws FirebaseAppDistributionException {
 
-    if (updateAppTaskCompletionSource != null
-        && !updateAppTaskCompletionSource.getTask().isComplete()) {
-      updateAppCancellationSource.cancel();
-    }
-
-    updateAppCancellationSource = new CancellationTokenSource();
-    updateAppTaskCompletionSource =
-        new TaskCompletionSource<>(updateAppCancellationSource.getToken());
-    Context context = firebaseApp.getApplicationContext();
-    this.updateTask = new UpdateTaskImpl(updateAppTaskCompletionSource.getTask());
+    if (isTesterSignedIn()) {
+      if (updateAppTaskCompletionSource != null
+          && !updateAppTaskCompletionSource.getTask().isComplete()) {
+        updateAppCancellationSource.cancel();
+      }
 
-    if (cachedLatestRelease == null) {
-      setUpdateAppTaskCompletionError(
-          new FirebaseAppDistributionException(
-              context.getString(R.string.no_update_available),
-              FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE));
-    } else {
-      if (cachedLatestRelease.getBinaryType() == BinaryType.AAB) {
-        redirectToPlayForAabUpdate(cachedLatestRelease.getDownloadUrl());
+      updateAppCancellationSource = new CancellationTokenSource();
+      updateAppTaskCompletionSource =
+          new TaskCompletionSource<>(updateAppCancellationSource.getToken());
+      Context context = firebaseApp.getApplicationContext();
+      this.updateTask = new UpdateTaskImpl(updateAppTaskCompletionSource.getTask());
+
+      if (cachedLatestRelease == null) {
+        setUpdateAppTaskCompletionError(
+            new FirebaseAppDistributionException(
+                context.getString(R.string.no_update_available),
+                FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE));
       } else {
-        // todo: create update class when implementing APK
-        throw new UnsupportedOperationException("Not yet implemented.");
+        if (cachedLatestRelease.getBinaryType() == BinaryType.AAB) {
+          redirectToPlayForAabUpdate(cachedLatestRelease.getDownloadUrl());
+        } else {
+          // todo: create update class when implementing APK
+          throw new UnsupportedOperationException("Not yet implemented.");
+        }
       }
+      return this.updateTask;
     }
 
-    return this.updateTask;
+    // if user is not signedIn
+    return new UpdateTaskImpl(
+        Tasks.forException(
+            new FirebaseAppDistributionException(
+                Constants.ErrorMessages.NOT_FOUND_ERROR,
+                FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE)));
```

Comment:
I know I suggested UPDATE NOT AVAILABLE earlier, but maybe this should be an authentication error instead